src/file-table-engine/src/table/immutable.rs

```diff
@@ -87,11 +112,37 @@ impl ImmutableFileTable {
         &self.metadata
     }
 
-    pub(crate) fn new(table_info: TableInfo, metadata: ImmutableMetadata) -> Self {
-        Self {
+    pub(crate) fn new(table_info: TableInfo, metadata: ImmutableMetadata) -> Result<Self> {
+        let table_info = Arc::new(table_info);
+        let options = table_info.meta.options.extra_options.clone();
```

Comment:
Maybe we don't need to clone this?
```suggestion
        let options = &table_info.meta.options.extra_options;
```